src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public aSide: number,
+    public bSide: number,
+    public cSide: number,
+  ) {
+    if (aSide <= 0 || bSide <= 0 || cSide <= 0) {
+      throw new Error('One of entered sides is less or equal 0');
+    }
+
+    if (
+      aSide + bSide <= cSide ||
+      aSide + cSide <= bSide ||
+      bSide + cSide <= aSide
+    ) {
+      throw new Error('Entered sides cannot create triangle');
```

Comment:
[CODE STYLE] - Similarly, provide a detailed error message for the triangle inequality check, such as 'The provided sides do not form a valid triangle according to the triangle inequality principle'.